joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/datastructures/RubyScope.scala

```diff
@@ -191,6 +196,74 @@ class RubyScope(summary: RubyProgramSummary, projectRoot: Option[String])
     }
   }
 
+  private def getNextClassTmp: String = {
+    val anonClassName = s"<anon-class-$tmpClassCounter>"
+    tmpClassCounter = tmpClassCounter + 1
+
+    anonClassName
+  }
+
+  private def getNextVarTmp: String = {
+    val tmpVarName = s"<tmp-$tmpVarCounter>"
+    tmpVarCounter = tmpVarCounter + 1
+
+    tmpVarName
+  }
+
+  private def getNextProcParamTmp: String = {
+    val procParamName = s"<proc-param-$tmpProcParamCounter>"
+    tmpProcParamCounter = tmpProcParamCounter + 1
+
+    procParamName
+  }
+
+  private def getNextClosureName: String = {
+    val closureName = s"<lambda>$tmpClosureCounter"
+    tmpClosureCounter = tmpClosureCounter + 1
+
+    closureName
+  }
+
+  def getNewClassTmp: String = {
+    val surroundingFullName = this.surroundingScopeFullName.getOrElse(RubyDefines.Main)
+
+    stack
+      .collectFirst {
+        case ScopeElement(scopeNode: NamespaceScope, _) => s"$surroundingFullName.${scopeNode.getNextClassTmp}"
+        case ScopeElement(node: TypeLikeScope, _)       => s"$surroundingFullName.${node.getNextClassTmp}"
+        case ScopeElement(node: MethodLikeScope, _)     => s"$surroundingFullName.${node.getNextClassTmp}"
+      }
+      .getOrElse(s"${surroundingFullName}.${this.getNextClassTmp}")
```

Comment:
Why do we need this fallback handling here and for all the other kinds of temp variables?